lib/config/dependencies.dart

```diff
@@ -2,14 +2,17 @@ import 'package:get_it/get_it.dart';
 import 'package:sidesail/providers/balance_provider.dart';
 import 'package:sidesail/routing/router.dart';
 import 'package:sidesail/rpc/rpc.dart';
+import 'package:sidesail/rpc/rpc_config.dart';
 import 'package:sidesail/storage/client_settings.dart';
 import 'package:sidesail/storage/secure_store.dart';
 
 // register all global dependencies, for use in views, or in view models
 // each dependency can only be registered once
 Future<void> initGetitDependencies() async {
+  // TODO: this can throw an error. How do we display that to the user?
```

Comment:
@octobocto do you know how to accomplish this?

Reply:
That's pretty hard. You would need to not throw an error, and store the error in the provider that fails. Then you can show the error message in the relevant component when the value is read

Reply:
Either store it in the provider that fails setup, or introduce another global provider that keeps track of initialization error messages

Reply:
Agree that crashing the whole app because we can't read the conf file is not the ideal solution 😛 this was mostly meant as a quick and dirty thing. 

The app does not function without this, though. So we should somehow display a big error message front and center, explaining to the user that we ran into an irrecoverable error

Reply:
quick n dirty is great, but (at some point)

we need some sort of component to show the error, and guide the user to solutions. Only way to do that is with a custom component

Reply:
I could set up a flow where we have an option to _not_ route to the home page if some sort of error occur. Would still have to save the error to the provider though